frontend/src/libs/packages/store/middlewares/handle-error.middleware.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  type AnyAction,
+  type Dispatch,
+  type Middleware,
+} from '@reduxjs/toolkit';
+import { isRejected } from '@reduxjs/toolkit';
+
+import { NotificationService } from '#libs/helpers/notification-service.js';
+
+const handleAction =
+  (next: Dispatch): ((action: AnyAction) => AnyAction) =>
+  (action: AnyAction): AnyAction => {
+    if (isRejected(action) && action.error.message) {
+      NotificationService.error(action.error.message);
+      return next(action);
+    }
+    return next(action);
+  };
+
+const handleError: Middleware = () => handleAction;
```

Comment:
Let's proceed with only handleError without handleAction

Reply:
Get dispatch form first callback param, add a type for it like AppDispatch(create this common type AppDispatch - typeof store.dispatch)